server/src/test/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitorTests.java

```diff
@@ -226,4 +206,262 @@ protected void markIndicesReadOnly(Set<String> indicesToMarkReadOnly, ActionList
+
+        assertTrue(clusterStateWithBlocks.blocks().indexBlocked(ClusterBlockLevel.WRITE, "test_2"));
+        AtomicLong currentTime = new AtomicLong();
+        DiskThresholdMonitor monitor = new DiskThresholdMonitor(Settings.EMPTY, () -> clusterStateWithBlocks,
+            new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS), null, currentTime::get, (reason, priority, listener) -> {
+            assertNotNull(listener);
+            assertThat(priority, equalTo(Priority.HIGH));
+            listener.onResponse(null);
+        }){
+            @Override
+            protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener<Void> listener, boolean readOnly) {
+                if (readOnly) {
+                    assertTrue(indicesToMarkReadOnly.compareAndSet(null, indicesToUpdate));
+                } else {
+                    assertTrue(indicesToRelease.compareAndSet(null, indicesToUpdate));
+                }
+                listener.onResponse(null);
+            }
+        };
+
+        // When free disk on any of node1 or node2 goes below 5% flood watermark, then apply index block on indices not having the block
+        indicesToMarkReadOnly.set(null);
+        indicesToRelease.set(null);
+        ImmutableOpenMap.Builder<String, DiskUsage> builder = ImmutableOpenMap.builder();
+        builder.put("node1", new DiskUsage("node1", "node1", "/foo/bar", 100, between(10, 100)));
+        builder.put("node2", new DiskUsage("node2", "node2", "/foo/bar", 100, between(0, 4)));
+        monitor.onNewInfo(new ClusterInfo(builder.build(), null, null, null));
+        assertEquals(new HashSet<>(Arrays.asList("test_1")), indicesToMarkReadOnly.get());
+        assertNull(indicesToRelease.get());
+
+        // When free disk on node1 and node2 goes above 10% high watermark, then only release index block
+        indicesToMarkReadOnly.set(null);
+        indicesToRelease.set(null);
+        builder = ImmutableOpenMap.builder();
+        builder.put("node1", new DiskUsage("node1", "node1", "/foo/bar", 100, between(10, 100)));
+        builder.put("node2", new DiskUsage("node2", "node2", "/foo/bar", 100, between(10, 100)));
+        monitor.onNewInfo(new ClusterInfo(builder.build(), null, null, null));
+        assertNull(indicesToMarkReadOnly.get());
+        assertEquals(new HashSet<>(Arrays.asList("test_2")), indicesToRelease.get());
+
+        //Remove index block settings from cluster state
+        indexMetaData = IndexMetaData.builder(clusterState.metaData().index("test_1")).settings(Settings.builder()
+            .put(clusterState.metaData()
+                .index("test_1").getSettings())
+            .put(IndexMetaData.INDEX_BLOCKS_READ_ONLY_ALLOW_DELETE_SETTING.getKey(), false)).build();
+
+        ClusterState clusterStateWithoutBlocks = ClusterState.builder(clusterState).metaData(MetaData.builder(clusterState.metaData())
```

Comment:
I don't understand this bit. What is it testing that hasn't already been covered?

Also, neither index has any blocks in `clusterState`, and `test_1` doesn't have any blocks in `clusterStateWithBlocks` either. Perhaps you mean to remove a block from `test_2`? But why not just use `clusterState` directly?